### ISSUE TITLE
building faces are missing

### ISSUE THREAD

**Issue by nvkelso** [mapzen/vector-datasource#402]
Testing dev against prod with exactly the same stylesheet removed 2 faces of extruded 3d buildings. 

This is a ship block for v0.5!

Is this related to the multipart change [we landed for MVT](https://github.com/mapzen/mapbox-vector-tile/issues/20)? When I set this to topojson (instead of mvt) it works fine.

sources:
    osm:
        # EraserMaps the app only supports MVT format tiles
        type: TopoJSONTiles #MVT
        url:   //vector.dev.mapzen.com/osm/all/{z}/{x}/{y}.topojson  #mvt

16/37.7857/-122.4014
https://github.com/tangrams/eraser-map

dev (v0.5-pre)
<img width="545" alt="screen shot 2015-11-10 at 15 16 04" src="https://cloud.githubusercontent.com/assets/853051/11079018/35716c9c-87be-11e5-985d-de61d9a671d4.png">

prod (v0.4)
<img width="581" alt="screen shot 2015-11-10 at 15 15 54" src="https://cloud.githubusercontent.com/assets/853051/11079019/358b8e92-87be-11e5-89bb-d1dfcf5b6e84.png">

**Comment by bcamper** [comment]
Whoa interesting. Is it possible winding order is backwards on that ring?
This can be a visual symptom of that. We enforce/fix winding order on
standalone/non-tiled data sources, but by default assume it is already
correct on tiled sources.

Can you try adding this option in the 'source' block, with the MVT source
and see if it makes any difference?

enforce_winding: true

**Comment by nvkelso** [comment]
So that totally fixes it, but what!?

Does that mean v0.5 tiles require Tangram v0.4 with this feature? That
seems weird. Is this a Tangram regression?

**Comment by bcamper** [comment]
No it's a tile data issue, it means you are probably right that it's from
that MVT change, which is apparently encoding some rings backwards.

**Comment by nvkelso** [comment]
@rmarianski over to you to debug as final preflight check for https://github.com/mapzen/mapbox-vector-tile/issues/20.

**Comment by zerebubuth** [comment]
This is really odd. According to the [Shapely docs](http://toblerity.org/shapely/manual.html#shapely.geometry.polygon.orient) the way we're using `orient` is correct:

> A sign of 1.0 means that the coordinates of the product’s exterior ring will be oriented counter-clockwise.

Which, with `y` positive, matches what [the v1.0.2 spec](https://github.com/mapbox/vector-tile-spec/pull/36/files#diff-3594e5e9052be0e33421df52071394d9R31) says about exterior rings:

> In a frame where the Y axis is positive up, this would make the winding order appear counter clockwise.

However, I just wrote [a little script](https://gist.github.com/zerebubuth/b5ffd64533992a889b9c) to check these things on dev, and it's erroring all over the place. Either one of these bits of documentation is wrong, or the "[Shoelace formula](https://en.wikipedia.org/wiki/Shoelace_formula)" produces opposite-signed areas to whatever method GEOS/Shapely is using. Confusement.

I'll see if just flipping the sign helps.

**Comment by zerebubuth** [comment]
I think mapzen/mapbox-vector-tile#22 will fix this issue.

The reason isn't that either the docs are wrong, or that the area calculations are different: It's that "a frame where the Y axis is positive up" does _not_ mean "geographic coordinate system", although that's how I interpreted it.

In fact, the `y` coordinates in an MVT tile are flipped from the geographic coordinate system, so a polygon that's in CCW orientation in geographic coordinates will continue to be CCW in the y-down tile coordinate system. The only problem with that is that we want the opposite orientation: clockwise orientation in geographic (y-up) coordinates, which will produce clockwise orientation in screen tile (y-down) coordinates, but counter-clockwise orientation in numeric tile coordinates and therefore a positive area.

TL;DR: before encoding polygons from a geographic coordinate system, they should be _clockwise_ oriented, which is the opposite of the usual OGC convention.

**Comment by zerebubuth** [comment]
I've confused myself so much that I've made a diagram to try and convince myself.

![tile_coords](https://cloud.githubusercontent.com/assets/271360/11091739/796ccdf8-8875-11e5-8708-8e5b901c5eeb.png)

**Comment by bcamper** [comment]
Why would we not have seen this problem before? A new processing step for
the multi-poly flattening?

**Comment by zerebubuth** [comment]
> Why would we not have seen this problem before? A new processing step for the multi-poly flattening?

Yup, we landed mapzen/mapbox-vector-tile#20, and that's what has caused this. We previously weren't enforcing an order, so it's possible that we are 99% right and 1% wrong in prod. That PR should have changed it to 100% right but, because confusedness, it looks like it as actually 100% wrong.

If there was a facepalm emoji, I would add it in here.

**Comment by nvkelso** [comment]
Thanks for investivating! Let's get this on dev to verify the fix.

**Comment by nvkelso** [comment]
Verified :)

<img width="674" alt="screen shot 2015-11-11 at 13 14 10" src="https://cloud.githubusercontent.com/assets/853051/11103246/254b316e-8876-11e5-8cc5-73209fd99fc0.png">
